### PR TITLE
Feature/516 enable typescript

### PR DESCRIPTION
**Pull request by danielsuter**
Enables typescript for karma and protractor tests.

Should be unproblematic for the release.

@adiherzog I guess some stuff is redundant with your branch.

### CODE REVIEW

File: scenarioo-client/package.json [hunk 1]
```diff
@@ -1,76 +1,83 @@
...
+    "ts-loader": "3.1.1",
+    "ts-node": "3.3.0",
+    "url-loader": "^0.5.9",
+    "webpack": "2.5.0",
+    "webpack-dev-server": "2.4.5",
+    "webpack-merge": "4.1.0"
+  },
+  "engines": {
+    "node": ">=4.4.0"
+  }
```
Comment (bruderol) [inline scenarioo-client/package.json]: @danielsuter please revert to old formatting - this change makes it not possible to see what really changed!
Seems like your formatting settings are wrong ?

File: scenarioo-client/prepareProtractor.js [hunk 2]
```diff
@@ -28,7 +32,7 @@ function prepareProtractorForE2ETests() {
         // this is also needed if scenarioo documenztation is not enabled
         var scenarioo = require('scenarioo-js');
         scenarioo.setupFluentDsl();
-        require('./test/protractorE2E/labelDefinitions');
+        require('./test/protractorE2E/labelDefinitions.js');
```
Comment (bruderol) [inline scenarioo-client/prepareProtractor.js]: the import should not be necessary to know whether it is a js or not

File: scenarioo-client/test/protractorE2E/specs/manage_branch_aliases.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 'use strict';
+import {scenario, step, useCase} from "scenarioo-js";
 
 var scenarioo = require('scenarioo-js');
```
Comment (bruderol) [inline scenarioo-client/test/protractorE2E/specs/manage_branch_aliases.ts]: we should as well migrate this requires to imports, shouldnt we?
why is this `scenarioo` used anyway here ??

Comment (bruderol) [reply scenarioo-client/test/protractorE2E/specs/manage_branch_aliases.ts]: dito for all other test files - maybe open an issue for future - if not resolvable immediately. Thanks.

File: scenarioo-client/test/protractorE2E/webPages/branchAliasesPage.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
 'use strict';
 
-var BaseWebPage = require('./baseWebPage.js'),
+import {by, element} from "protractor";
+
+var BaseWebPage = require('./baseWebPage'),
```
Comment (bruderol) [inline scenarioo-client/test/protractorE2E/webPages/branchAliasesPage.ts]: cant we use import for that one?

File: scenarioo-client/prepareProtractor.js (hunk 2 above)
Comment (bruderol) [reply scenarioo-client/prepareProtractor.js]: concerning this not working properly with typescript: please open an issue on scenariooJS repo, might be something missing in the typings. eat your own dog food rules ;-)

**Comment by bruderol** [comment]
I think it is great that we can use type script in our tests - one thing that I want to question: does it make sense to rename js files to ts without really migrating them to real typescript - like that we are hiding which files have been thoughtfully migrated to use appropriate types. I would rather configure it in such a way, that javascript files can still be used as tests.

File: scenarioo-client/test/protractorE2E/webPages/branchAliasesPage.ts (hunk 4 above)
Comment (danielsuter) [reply scenarioo-client/test/protractorE2E/webPages/branchAliasesPage.ts]: don't know how

File: scenarioo-client/test/protractorE2E/specs/manage_branch_aliases.ts (hunk 3 above)
Comment (danielsuter) [reply scenarioo-client/test/protractorE2E/specs/manage_branch_aliases.ts]: dead import

**Comment by danielsuter** [comment]
@bruderol Thanks for your feedback.
I made the changes which were easy. I'm not going to refactor all our code, since this is not really part of my Angular 5 migration. That will be a team effort.

**Comment by bruderol** [comment]
@danielsuter I think you did not understand me. I was not saying that you have to migrate everything! Of ocurse not. And I totaly agree that we all have to help with this! My point was just: please not rename javascript files to ts without really migrating the content, such that we see which files have been properly migrated.

File: scenarioo-client/test/protractorE2E/specs/manage_branch_aliases.ts [hunk 5]
```diff
@@ -1,7 +1,6 @@
 'use strict';
 import {scenario, step, useCase} from "scenarioo-js";
 
-var scenarioo = require('scenarioo-js');
 var pages = require('./../webPages');
```
Comment (bruderol) [inline scenarioo-client/test/protractorE2E/specs/manage_branch_aliases.ts]: @danielsuter did you open an issue for getting rid of the requires in all tests? also the unused scenarioo is somehow in all tests --> open an issue for we not forget to clean this up. Thanks a lot!

Comment (danielsuter) [reply scenarioo-client/test/protractorE2E/specs/manage_branch_aliases.ts]: @bruderol  Thanks for the reminder, I opened #652

**Pull request merged**